### ISSUE TITLE
TestTxnWriteFail: timed out waiting for get

### ISSUE THREAD

**Issue by heyitsanthony** [coreos/etcd#6238]
via semaphore (https://semaphoreci.com/coreos/etcd/branches/master/builds/817)
```
--- FAIL: TestTxnWriteFail (5.40s)
	txn_test.go:94: timed out waiting for get
	txn_test.go:84: grpc: the client connection is closing
```

**Comment by xiang90** [comment]
@heyitsanthony After adding the grep thing into the CI, I checked a few failed cases. This issues caused a few failures.

**Comment by heyitsanthony** [comment]
@xiang90 yesterday @gyuho debugged it down to a leader election in #6260 but `Get` shouldn't be returning a connection closing error

**Comment by gyuho** [comment]
Yeah it seems like gRPC returning unexpected ErrClientConnClosing. I will investigate further once Anthony's PR is merged.

**Comment by heyitsanthony** [comment]
Fixed by #6264
